modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotCheckProcess.java

```diff
@@ -474,70 +692,100 @@ private <T> GridFutureAdapter<T> phaseFuture() {
         }
     }
 
-    /** A DTO used to transfer nodes' results for the both phases. */
-    private static final class SnapshotCheckResponse implements Serializable {
-        /** Serial version uid. */
-        private static final long serialVersionUID = 0L;
-
-        /** @see #metas() */
-        @Nullable private final List<SnapshotMetadata> metas;
+    /** A DTO base to transfer node's results for the both phases. */
+    private abstract static class AbstractSnapshotCheckResponse<T> implements Serializable {
```

Comment:
Why this class abstract, when all inheritors just use constructor. Why do you need all this classes?

Reply:
To avoid plenty of large generics in code like 'SnapshotCheckResponse<Map<String, Map<PartitionKeyV2, PartitionHashRecordV2>>>'.